### PR TITLE
misc (non-functional) changes to cudnn backend

### PR DESCRIPTION
**Pull request by ankan-ban**
- Replace all cudaMemcpyAsync used for loading weights with cudaMemcpy as  source (in CPU memory) could be deleted before the async version of the function actually does the copy. Thanks @Tilps for pointing this out.
- Add comments explaining what the policy map layer does and how the layout conversion from CHW to HWC works.
- replace new/delete with std::vector for the allocation of scratch space used by layout conversion.
- minor naming/style change.

### CODE REVIEW

File: src/neural/cuda/layers.cc [hunk 1]
```diff
@@ -300,24 +300,23 @@ void SELayer<float>::LoadWeights(float* w1, float* b1, float* w2, float* b2,
...
   // Weight for the second FC layer.
-  ReportCUDAErrors(
-      cudaMemcpyAsync(w2_, w2, weight_size2, cudaMemcpyHostToDevice));
+  ReportCUDAErrors(cudaMemcpy(w2_, w2, weight_size2, cudaMemcpyHostToDevice));
 
   // Bias for the first FC layer.
-  ReportCUDAErrors(cudaMemcpyAsync(b1_, b1, numFc1Out_ * sizeof(float),
+  ReportCUDAErrors(
+      cudaMemcpy(b1_, b1, numFc1Out_ * sizeof(float),
                                    cudaMemcpyHostToDevice));
```
Comment (Ttl) [inline src/neural/cuda/layers.cc]: Some of these lines could use clang-format.

**Comment by Ttl** [comment]
I think that `CudaMemcpyAsync` needs the CPU buffer to be page-locked memory for it to be able to copy asynchronously. Since none of these buffers are page-locked, copies are actually synchronous and there wasn't risk of them failing. Using `cudaMemcpy` instead is still a good change to make it explicit.

**Comment by ankan-ban** [comment]
Agree, that was likely the reason `CudaMemcpyAsync` was working. The documentation isn't very clear though so I agree that we should change to `CudaMemcpy` to be safe in case some later version of cuda stops supporting it.

File: src/neural/cuda/layers.cc [hunk 2]
```diff
@@ -591,13 +636,12 @@ void PolicyMapLayer<DataType>::LoadWeights(const short* cpuWeight,
         else
           convertedWeights[hw * Cin + c] = -1;
       }
-    ReportCUDAErrors(cudaMemcpyAsync(weights_, convertedWeights,
-                                     usedSize * sizeof(short),
-                                     cudaMemcpyHostToDevice));
-    delete[] convertedWeights;
+    ReportCUDAErrors(cudaMemcpy(weights_, &convertedWeights[0],
```
Comment (Tilps) [inline src/neural/cuda/layers.cc]: convertedWeights.data() ?

**Pull request merged**